src/commands/api/create.js

```diff
@@ -23,47 +23,54 @@ class CreateAPICommand extends BaseCommand {
     })
   }
 
-  async tryCreateApi({ flags, apiPath, oas, versionToCreate }) {
-    const isNameAvailable = await this.checkApiName(apiPath)
+  async tryCreateApi({ path, version, oas, flags }) {
+    const isNameAvailable = await this.checkApiName(path)
+    const pathHasVersion = path.length === 3
+    const fullPath = pathHasVersion ? path : [...path, version]
     
     if (isNameAvailable) {
-      const [owner, name, version = versionToCreate] = apiPath
-      return this.createApi(owner, name, version, oas, flags, successMessage(apiPath))
-        .then(() => true)
+      await this.createApi(fullPath, oas, flags, successMessage(path))
+      return true
     }
     
-    return Promise.resolve(isNameAvailable)
+    return isNameAvailable
   }
 
-  async tryCreateApiVersion({ apiPath, version, ...args }) {
-    return this.tryCreateApi({ ...args, apiPath: [...apiPath, version] })
+  async tryCreateApiVersion({ path, ...args }) {
+    return this.tryCreateApi({ 
+      path: [...path, args.version],
+      ...args
+    })
   }
 
   async run() {
     const { args, flags } = this.parse(CreateAPICommand)
-    const [owner, name, version] = getApiIdentifierArg(args, false).split('/')
     const definition = parseDefinition(flags.file)
-    const oas = getOasVersion(definition)
-    const versionToCreate = version || getVersion(definition)
+    const [oas, apiVersion] = from(definition)(getOasVersion, getVersion)
+    const requestedApiPath = getApiIdentifierArg(args, false)
+    const [owner, name, version = apiVersion] = splitPathParams(requestedApiPath)
 
     const argsObj = {
+      path: [owner, name],
+      version,
       flags,
-      apiPath: [owner, name],
-      oas, 
-      versionToCreate 
+      oas
     }
 
-    return (
-      await this.tryCreateApi(argsObj) ||
-      await this.tryCreateApiVersion({ ...argsObj, version: versionToCreate }) ||
-      this.error(
-        errorMsg.apiVersionExists({ owner, name, versionToCreate }), 
+    const createdApi = (
+      await this.tryCreateApi(argsObj) || 
+      await this.tryCreateApiVersion(argsObj)
+    )
+
+    if (!createdApi) {
+      return this.error(
+        errorMsg.apiVersionExists({ owner, name, version }), 
         { exit: 1 }
       )
-    )
+    }
   }
 
-  async createApi(owner, name, version, oas, flags, successMessage) {
+  async createApi([owner, name, version], oas, flags, successMessage) {
```

Comment:
While not a direct change from this PR, would moving this function to before `run()` improve the readability? 

This is probably a little pedantic, but I struggled with reasoning this command slightly. It feels like this function is out of place and that it belongs alongside the other command helper functions (`tryCreateApiVersion`)

Reply:
Yep. I think I that makes sense. Will update